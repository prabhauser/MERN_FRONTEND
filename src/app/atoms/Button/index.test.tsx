import { shallow, mount } from 'enzyme';
import ADBButton from './index';

describe('<ADBButton />', () => {
    it('Expects to add a button in to the DOM', () => {
        const wrapper = shallow(<ADBButton name="test" />);
        expect(wrapper.exists()).toBe(true);
    });

    it('Check the length of the DOM', () => {
        const wrapper = shallow(<ADBButton name="test" />);
        expect(wrapper.length).toEqual(1);
    });

    it('Expects to find button HTML element with className test in the DOM', () => {
        const wrapper = mount(
            <ADBButton
                secondary={false}
                isShowEndIcon={false}
                formInput={'test'}
                fullWidth={true}
                isShowStartIcon={false}
                disabled={true}
                name="test"
                className="test"
                type="submit"
                StartIconArrow={'test'}
            />
        );
        expect(wrapper.find('button.test')).toHaveLength(1);
    });
    it('should take Button snapshot', () => {
        const component = shallow(<ADBButton name="test" className="test" type="submit" />);
        component.update();
        expect(component).toMatchSnapshot();
    });
});
