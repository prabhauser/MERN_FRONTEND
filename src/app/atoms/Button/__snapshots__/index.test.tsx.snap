// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ADBButton /> should take Button snapshot 1`] = `
<div>
  <WithStyles(ForwardRef(Button))
    className=" makeStyles-root-1 makeStyles-primaryButton-2 test"
    disabled={false}
    name="test"
    type="submit"
  >
    test
  </WithStyles(ForwardRef(Button))>
</div>
`;
